### PR TITLE
Updated CLI token source `parseExpiry` method to account for different time format

### PR DESCRIPTION
**Pull request by tanmay-db**
## Changes

Add functionality to check for different date format for expiry parsing in CLI token source. Also add pattern matching for flexibility between different java versions

## Tests

Added unit tests, since `LocalDateTime` support upto nano second precision -- tested for less and more digits as well.

### CODE REVIEW

File: databricks-sdk-java/src/main/java/com/databricks/sdk/core/CliTokenSource.java [hunk 1]
```diff
@@ -37,10 +37,13 @@ public CliTokenSource(
     this.getAllEnv = getAllEnv;
   }
 
-  private static LocalDateTime parseExpiry(String expiry) {
+  static LocalDateTime parseExpiry(String expiry) {
```
Comment (tanmay-db) [inline databricks-sdk-java/src/main/java/com/databricks/sdk/core/CliTokenSource.java:40]: Making package private

**Review by cdkrot — APPROVED** [review]
This looks good to me. May be we can also add test into cli to assert those are the only formats it can output?

**Comment by tanmay-db** [comment]
> This looks good to me. May be we can also add test into cli to assert those are the only formats it can output?

Not sure if we can add test into az cli since we don't build it. Probably in the databricks cli we can (not sure if there already is) but that would be external of Java SDK.

**Comment by cdkrot** [comment]
Into *databricks cli I mean

The problem I have reported on slack was with our cli, not az cli (but there are a LOT problems with az cli as well)

**Comment by tanmay-db** [comment]
> Into *databricks cli I mean
> 
> The problem I have reported on slack was with our cli, not az cli (but there are a LOT problems with az cli as well)
cc: @pietern for visibility on CLI

**Review by nfx — APPROVED** [review]

**Pull request merged**